src/Mod/Part/Gui/ViewProviderMirror.cpp

```diff
@@ -373,17 +374,18 @@ void ViewProviderChamfer::updateData(const App::Property* prop)
 
             auto vpBase = dynamic_cast<PartGui::ViewProviderPart*>(Gui::Application::Instance->getViewProvider(objBase));
             if (vpBase) {
-                std::vector<App::Color> colBase = static_cast<PartGui::ViewProviderPart*>(vpBase)->DiffuseColor.getValues();
-                std::vector<App::Color> colCham;
-                colCham.resize(chamMap.Extent(), static_cast<PartGui::ViewProviderPart*>(vpBase)->ShapeAppearance.getDiffuseColor());
-                applyTransparency(static_cast<PartGui::ViewProviderPart*>(vpBase)->Transparency.getValue(),colBase);
-
-                if (static_cast<int>(colBase.size()) == baseMap.Extent()) {
-                    applyColor(hist[0], colBase, colCham);
+                // std::vector<App::Color> colBase = static_cast<PartGui::ViewProviderPart*>(vpBase)->DiffuseColor.getValues();
+                auto& colBase = static_cast<PartGui::ViewProviderPart*>(vpBase)->ShapeAppearance;
+                std::vector<App::Material> colCham;
+                colCham.resize(chamMap.Extent(), colBase[0]);
+                colBase.setTransparency(static_cast<PartGui::ViewProviderPart*>(vpBase)->Transparency.getValue());
+
+                if (static_cast<int>(colBase.getSize()) == baseMap.Extent()) {
+                    applyMaterial(hist[0], colBase, colCham);
                 }
-                else if (!colBase.empty() && colBase[0] != this->ShapeAppearance.getDiffuseColor()) {
-                    colBase.resize(baseMap.Extent(), colBase[0]);
-                    applyColor(hist[0], colBase, colCham);
+                else if (colBase.getSize() > 0 && colBase[0] != this->ShapeAppearance[0]) {

```

Comment:
Why change to `getSize() > 0`? I typically advise the opposite change (though I wasn't going to bother suggesting it in this PR).

Reply:
colBase is no longer a vector, so there is no empty function